infra/modules/providers/azure/service-principal/sample/unit_test.go

```diff
@@ -0,0 +1,74 @@
+//  Copyright © Microsoft Corporation
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package test
+
+import (
+	"encoding/json"
+	"testing"
+
+	"github.com/gruntwork-io/terratest/modules/random"
+	"github.com/gruntwork-io/terratest/modules/terraform"
+	"github.com/microsoft/cobalt/test-harness/infratests"
+)
+
+var name = "serviceprincipal-"
+var count = 7
+
+var tfOptions = &terraform.Options{
+	TerraformDir: "./",
+	Upgrade:      true,
+}
+
+func asMap(t *testing.T, jsonString string) map[string]interface{} {
+	var theMap map[string]interface{}
+	if err := json.Unmarshal([]byte(jsonString), &theMap); err != nil {
+		t.Fatal(err)
+	}
+	return theMap
+}
+
+func TestTemplate(t *testing.T) {
+
+	expectedResult := asMap(t, `{
+		"available_to_other_tenants": false,
+		"type": "webapp/api",
+		"required_resource_access": [{
+			"resource_app_id": "00000003-0000-0000-c000-000000000000",
+			"resource_access": [{
+				"id": "06da0dbc-49e2-44d2-8312-53f166ab848a",
```

Comment:
What are these resources? "06da0dbc-49e2-44d2-8312-53f166ab848a"

Reply:
The Unit Test is testing that the Service Principal should have a Permission Role access of Directory.Read.All to the Microsoft Graph

Microsoft Graph	https://graph.microsoft.com  =  00000003-0000-0000-c000-000000000000
Permission Role ID Directory.Read.All = 06da0dbc-49e2-44d2-8312-53f166ab848a

Reference:  https://docs.microsoft.com/en-us/graph/permissions-reference